File: frontend/src/services/equipmentService.ts
import { apiClient } from '@/config/apiClient';
import type { 
  ApiResponse, 
  PageResponse, 
  EquipmentResponse, 
  EquipmentRequest, 
  TableParams 
} from '@/types';

export interface EquipmentTableParams extends TableParams {
  type?: string;
  status?: string;
}

export const equipmentService = {
  getAll: (params?: EquipmentTableParams): Promise<ApiResponse<PageResponse<EquipmentResponse>>> => {
    return apiClient.get('/equipment', { params });
  },

  getById: (id: number): Promise<ApiResponse<EquipmentResponse>> => {
    return apiClient.get(`/equipment/${id}`);
  },

  create: (data: EquipmentRequest): Promise<ApiResponse<EquipmentResponse>> => {
    return apiClient.post('/equipment', data);
  },

  update: (id: number, data: EquipmentRequest): Promise<ApiResponse<EquipmentResponse>> => {
    return apiClient.put(`/equipment/${id}`, data);
  },

  delete: (id: number): Promise<ApiResponse<void>> => {
    return apiClient.delete(`/equipment/${id}`);
  },
};

File: backend/src/main/java/com/bubbleflow/service/EquipmentService.java
package com.bubbleflow.service;

import com.bubbleflow.dto.equipment.request.EquipmentRequest;
import com.bubbleflow.dto.equipment.response.EquipmentResponse;
import com.bubbleflow.dto.response.PageResponse;
import org.springframework.data.domain.Pageable;

public interface EquipmentService {
    PageResponse<EquipmentResponse> getAll(String search, String type, String status, Pageable pageable);
    EquipmentResponse getById(Long id);
    EquipmentResponse create(EquipmentRequest request);
    EquipmentResponse update(Long id, EquipmentRequest request);
    void delete(Long id);
}
